Full Yellow Local - WazuhDB tests in 4.2

## Description

This issue is part of the https://github.com/wazuh/wazuh-qa/issues/1516. After the research #1493, it has been detected that  `wazuh_db` tests do not properly autoconfigure the testing environment. These tests do not start automatically `wazuh_db` daemon. In an environment where the Wazuh manager has not started for the first time, these tests will fail in some cases.

##  Research results

### Pipeline parameters

|  Jenkins branch  | QA branch   | 
|- |- |
|  4.2 |   1516-4.2.0-full-green |

### Packages details

Type | Format | Architecture | Version | Revision | Tag | File name
-- | -- | -- | -- | -- | -- | --
Server | rpm | x86_64 | 4.2.0 | 40207 | v4.2.0-rc8 | wazuh-manager-4.2.0-1.1493.x86_64.rpm


In the case presented in the description, wazuh_db tests fail in the following cases:

```

FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: Insert commands] - AssertionErro...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: Update commands] - AssertionErro...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: Select commands] - AssertionErro...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: get-all-agents command] - Assert...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: sync-agent-info-get command] - A...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: Reset connection status] - Asser...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: get-agents-by-connection-status command]
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: disconnect-agents command] - Ass...
===================================== 8 failed, 22 passed in 26.11s =====================================
```

The error output obtained is:

```
test_wazuh_db/test_wazuh_db.py ...............FF.FFF..FFF....                                     [100%]
=============================================== FAILURES ================================================
____________________________ test_wazuh_db_messages[global: Insert commands] ____________________________
configure_sockets_environment = None
connect_to_sockets_module = [<wazuh_testing.tools.monitoring.SocketController object at 0x7f7b39d165f8>]
test_case = [{'input': 'global delete-group TestGroup1', 'output': 'ok', 'stage': 'global delete-group previous insert'}, {'input'...lect-groups', 'output': 'ok [{"name":"default"},{"name":"TestGroup1"}]', 'stage': 'global select-groups success'}, ...]
    @pytest.mark.parametrize('test_case',
                             [case['test_case'] for module_data in module_tests for case in module_data[0]],
                             ids=[f"{module_name}: {case['name']}"
                                  for module_data, module_name in module_tests
                                  for case in module_data]
                             )
    def test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_module, test_case: list):
        """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
        Parameters
        ----------
        test_case : list
            List of test_case stages (dicts with input, output and stage keys).
        """
        for index, stage in enumerate(test_case):
            if 'ignore' in stage and stage['ignore'] == "yes":
                continue
            expected = stage['output']
            receiver_sockets[0].send(stage['input'], size=True)
            response = receiver_sockets[0].receive(size=True).decode()
            if 'use_regex' in stage and stage['use_regex'] == 'yes':
                match = True if regex_match(expected, response) else False
            else:
                match = (expected == response)
>           assert match, 'Failed test case stage {}: {}. Expected: {}. Response: {}' \
                .format(index + 1, stage['stage'], expected, response)
E           AssertionError: Failed test case stage 4: global find-group after insert. Expected: ok [{"id":*}]. Response: ok []
E           assert False
test_wazuh_db/test_wazuh_db.py:105: AssertionError
____________________________ test_wazuh_db_messages[global: Update commands] ____________________________
```

All tests should autoconfigure the testing environment in order to prevent unexpected behaviours.

It has been detected that this test fails in case of the manager has already registered some agents before the test is run. This should be fixed in other to guarantee independence between tests.


"""
test_wazuh_db/test_wazuh_db.py ...................F.F.F......                                                                                                                                               [100%]

==================================================================================================== FAILURES =====================================================================================================
_____________________________________________________________________________ test_wazuh_db_messages[global: get-all-agents command] ______________________________________________________________________________

configure_sockets_environment = None, connect_to_sockets_module = [<wazuh_testing.tools.monitoring.SocketController object at 0x7f5495e72668>]
test_case = [{'input': 'global get-all-agents last_id -1', 'output': 'ok [{"id":0},{"id":1},{"id":2}]', 'stage': 'global get-all-agents success'}]

    @pytest.mark.parametrize('test_case',
                             [case['test_case'] for module_data in module_tests for case in module_data[0]],
                             ids=[f"{module_name}: {case['name']}"
                                  for module_data, module_name in module_tests
                                  for case in module_data]
                             )
    def test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_module, test_case: list):
        """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
    
        Parameters
        ----------
        test_case : list
            List of test_case stages (dicts with input, output and stage keys).
        """
        for index, stage in enumerate(test_case):
            if 'ignore' in stage and stage['ignore'] == "yes":
                continue
    
            expected = stage['output']
            receiver_sockets[0].send(stage['input'], size=True)
            response = receiver_sockets[0].receive(size=True).decode()
    
            if 'use_regex' in stage and stage['use_regex'] == 'yes':
                match = True if regex_match(expected, response) else False
            else:
                match = (expected == response)
>           assert match, 'Failed test case stage {}: {}. Expected: {}. Response: {}' \
                .format(index + 1, stage['stage'], expected, response)
E           AssertionError: Failed test case stage 1: global get-all-agents success. Expected: ok [{"id":0},{"id":1},{"id":2}]. Response: ok [{"id":0},{"id":1},{"id":2},{"id":3}]
E           assert False

test_wazuh_db/test_wazuh_db.py:105: AssertionError
___________________________________________________________________________ test_wazuh_db_messages[global: sync-agent-info-set command] ___________________________________________________________________________

configure_sockets_environment = None, connect_to_sockets_module = [<wazuh_testing.tools.monitoring.SocketController object at 0x7f5495e72668>]
test_case = [{'input': 'global sync-agent-info-set [{"id":3,"name":"TestName3","ip":"0.0.0.3","os_name":"TestOsName","os_version":...,"connection_status":"never_connected"}]', 'stage': 'global get-agent-info success after sync-agent-info-set success'}]

    @pytest.mark.parametrize('test_case',
                             [case['test_case'] for module_data in module_tests for case in module_data[0]],
                             ids=[f"{module_name}: {case['name']}"
                                  for module_data, module_name in module_tests
                                  for case in module_data]
                             )
    def test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_module, test_case: list):
        """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
    
        Parameters
        ----------
        test_case : list
            List of test_case stages (dicts with input, output and stage keys).
        """
        for index, stage in enumerate(test_case):
            if 'ignore' in stage and stage['ignore'] == "yes":
                continue
    
            expected = stage['output']
            receiver_sockets[0].send(stage['input'], size=True)
            response = receiver_sockets[0].receive(size=True).decode()
    
            if 'use_regex' in stage and stage['use_regex'] == 'yes':
                match = True if regex_match(expected, response) else False
            else:
                match = (expected == response)
>           assert match, 'Failed test case stage {}: {}. Expected: {}. Response: {}' \
                .format(index + 1, stage['stage'], expected, response)
E           AssertionError: Failed test case stage 1: global sync-agent-info-set non-existent agent. Expected: ok. Response: err Cannot execute Global database query; NOT NULL constraint failed: agent.connection_status
E           assert False

test_wazuh_db/test_wazuh_db.py:105: AssertionError
_____________________________________________________________________________ test_wazuh_db_messages[global: Reset connection status] _____________________________________________________________________________

configure_sockets_environment = None, connect_to_sockets_module = [<wazuh_testing.tools.monitoring.SocketController object at 0x7f5495e72668>]
test_case = [{'input': 'global update-connection-status {"id":1,"connection_status":"pending","sync_status":"synced"}', 'output': ...":"active"}]', 'stage': 'global manager get-agent-info success post reset connection status', 'use_regex': 'yes'}, ...]

    @pytest.mark.parametrize('test_case',
                             [case['test_case'] for module_data in module_tests for case in module_data[0]],
                             ids=[f"{module_name}: {case['name']}"
                                  for module_data, module_name in module_tests
                                  for case in module_data]
                             )
    def test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_module, test_case: list):
        """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
    
        Parameters
        ----------
        test_case : list
            List of test_case stages (dicts with input, output and stage keys).
        """
        for index, stage in enumerate(test_case):
            if 'ignore' in stage and stage['ignore'] == "yes":
                continue
    
            expected = stage['output']
            receiver_sockets[0].send(stage['input'], size=True)
            response = receiver_sockets[0].receive(size=True).decode()
    
            if 'use_regex' in stage and stage['use_regex'] == 'yes':
                match = True if regex_match(expected, response) else False
            else:
                match = (expected == response)
>           assert match, 'Failed test case stage {}: {}. Expected: {}. Response: {}' \
                .format(index + 1, stage['stage'], expected, response)
E           AssertionError: Failed test case stage 9: global agent get-agent-info success post reset connection status. Expected: ok [{"id":3,"name":"TestName3","ip":"0.0.0.3","register_ip":"0.0.0.3","internal_key":"TopSecret","os_name":"TestOsName","os_version":"TestOsVersion","os_major":"TestOsMajor","os_minor":"TestOsMinor","os_codename":"TestOsCodeName","os_build":"TestOsBuild","os_platform":"TestOsPlatfor","os_uname":"TestOsUname","os_arch":"TestOsArch","config_sum":"TestConfigSum","merged_sum":"TestMergedSum","manager_host":"TestManagerHost","node_name":"TestNodeName","date_add":1599223378,"last_keepalive":200,"sync_status":"syncreq","connection_status":"disconnected"}]. Response: ok [{"id":3,"name":"1-b42dc306-debian7","ip":"10.0.2.15","register_ip":"any","internal_key":"71cf956218f4e5c839d538a6ed09ce81b9152e7b4e19132984ee19fac988ebda","os_name":"Debian GNU/Linux","os_version":"7","os_major":"7","os_codename":"wheezy","os_platform":"debian","os_uname":"Linux |agent-debian7 |3.16.0-9-amd64 |#1 SMP Debian 3.16.68-1 (2019-05-22) |x86_64","os_arch":"x86_64","version":"Wazuh v3.12.0","config_sum":"ab73af41699f13fdd81903b5f23d8d00","merged_sum":"2c45c95db2954d2c7d0ea533f09e81a5","manager_host":"centos1","node_name":"node01","date_add":1626728461,"last_keepalive":1626728531,"group":"default","sync_status":"syncreq","connection_status":"disconnected"}]
E           assert False

test_wazuh_db/test_wazuh_db.py:105: AssertionError
============================================================================================= short test summary info =============================================================================================
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: get-all-agents command] - AssertionError: Failed test case stage 1: global get-all-agents success. Expected: ok [{"id":0},{"id":1},{"id":2...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: sync-agent-info-set command] - AssertionError: Failed test case stage 1: global sync-agent-info-set non-existent agent. Expected: ok. Resp...
FAILED test_wazuh_db/test_wazuh_db.py::test_wazuh_db_messages[global: Reset connection status] - AssertionError: Failed test case stage 9: global agent get-agent-info success post reset connection status. Exp...
========================================================================================== 3 failed, 27 passed in 34.14s ==========================================================================================
"""

--- a/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
@@ -3,36 +3,29 @@ import subprocess
 import subprocess as sb
 from wazuh_testing.tools import WAZUH_PATH
 from wazuh_testing.wazuh_db import query_wdb
-import sqlite3
 
 
-def list_agents(fields=['id']):
-    global_db_conn = sqlite3.connect(f"{WAZUH_PATH}/queue/db/global.db")
-    selected_fields = ''.join(fields)
-    global_db_conn.row_factory = lambda cursor, row: row[0]
-    cursor = global_db_conn.cursor()
 
-    list_agents = []
-
-    try:
-        sqlite_response = cursor.execute(f"SELECT id FROM agent").fetchall()
-        sqlite_response.remove(0)
-        list_agents = sqlite_response 
-    except Exception:
-        list_agents = []
+def list_agents_ids():
+    wazuhdb_result = query_wdb("global get-all-agents last_id -1")
+    list_agents = [agent['id'] for agent in wazuhdb_result if not (0 == agent.get('id'))]
 
     return list_agents
 
 
-def remove_agents(agents_id, wazuh_script=True, wazuh_db_query=False):
+def remove_agents(agents_id, remove_option='wazuhdb'):
+    if remove_option not in ['wazuhdb','manage_agents']:
+        raise ValueError("Invalid type of agent removal: %s" % remove_option)
+
     if agents_id:
         for agent_id in agents_id:
-            if wazuh_script:
+            if remove_option == 'manage_agents':
                 subprocess.call([f"{WAZUH_PATH}/bin/manage_agents", "-r", f"{agent_id}"], stdout=open(os.devnull, "w"),
                                 stderr=subprocess.STDOUT)
-            if wazuh_db_query:
+            elif remove_option == 'wazuhdb':
                 result = query_wdb(f"global delete-agent {str(agent_id)}")
 
+
 def remove_all_agents():
-    agent_ids_list = list_agents(['id'])
+    agent_ids_list = list_agents_ids()
     remove_agents(agent_ids_list)

--- a/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
@@ -5,7 +5,6 @@ from wazuh_testing.tools import WAZUH_PATH
 from wazuh_testing.wazuh_db import query_wdb
 
 
-
 def list_agents_ids():
     wazuhdb_result = query_wdb("global get-all-agents last_id -1")
     list_agents = [agent['id'] for agent in wazuhdb_result if not (0 == agent.get('id'))]
@@ -13,19 +12,19 @@ def list_agents_ids():
     return list_agents
 
 
-def remove_agents(agents_id, remove_option='wazuhdb'):
-    if remove_option not in ['wazuhdb','manage_agents']:
-        raise ValueError("Invalid type of agent removal: %s" % remove_option)
+def remove_agents(agents_id, remove_type='wazuhdb'):
+    if remove_type not in ['wazuhdb','manage_agents']:
+        raise ValueError("Invalid type of agent removal: %s" % remove_type)
 
     if agents_id:
         for agent_id in agents_id:
-            if remove_option == 'manage_agents':
+            if remove_type == 'manage_agents':
                 subprocess.call([f"{WAZUH_PATH}/bin/manage_agents", "-r", f"{agent_id}"], stdout=open(os.devnull, "w"),
                                 stderr=subprocess.STDOUT)
-            elif remove_option == 'wazuhdb':
+            elif remove_type == 'wazuhdb':
                 result = query_wdb(f"global delete-agent {str(agent_id)}")
 
 
-def remove_all_agents():
+def remove_all_agents(remove_type):
     agent_ids_list = list_agents_ids()
-    remove_agents(agent_ids_list)
+    remove_agents(agent_ids_list, remove_type)

--- a/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
@@ -1,0 +1,32 @@
+import os
+import subprocess
+import subprocess as sb
+from wazuh_testing.tools import WAZUH_PATH
+from wazuh_testing.wazuh_db import query_wdb
+import sqlite3
+import time
+
+
+def list_agents(fields=['id']):
+    global_db_conn = sqlite3.connect(f"{WAZUH_PATH}/queue/db/global.db")
+    selected_fields = ''.join(fields)
+    global_db_conn.row_factory = lambda cursor, row: row[0]
+    cursor = global_db_conn.cursor()
+    
+    sqlite_response = cursor.execute(f"SELECT id FROM agent").fetchall()
+    sqlite_response.remove(0)
+    return sqlite_response 
+
+
+def remove_agents(agents_id):
+    print(f"Removing {agents_id}")
+
+    if agents_id:
+        for agent_id in agents_id:
+           print(f"Removing {str(agent_id)}")
+           subprocess.call([f"{WAZUH_PATH}/bin/manage_agents", "-r", f"{agent_id}"], stdout=open(os.devnull, "w"), stderr=subprocess.STDOUT)
+           #result =  query_wdb(f"global delete-agent {str(agent_id)}")
+
+def remove_all_agents():
+    agent_ids_list = list_agents(['id'])
+    remove_agents(agent_ids_list)

--- a/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
@@ -4,7 +4,6 @@ import subprocess as sb
 from wazuh_testing.tools import WAZUH_PATH
 from wazuh_testing.wazuh_db import query_wdb
 import sqlite3
-import time
 
 
 def list_agents(fields=['id']):
@@ -12,20 +11,27 @@ def list_agents(fields=['id']):
     selected_fields = ''.join(fields)
     global_db_conn.row_factory = lambda cursor, row: row[0]
     cursor = global_db_conn.cursor()
-    
-    sqlite_response = cursor.execute(f"SELECT id FROM agent").fetchall()
-    sqlite_response.remove(0)
-    return sqlite_response 
+
+    list_agents = []
+
+    try:
+        sqlite_response = cursor.execute(f"SELECT id FROM agent").fetchall()
+        sqlite_response.remove(0)
+        list_agents = sqlite_response 
+    except Exception:
+        list_agents = []
+
+    return list_agents
 
 
-def remove_agents(agents_id):
-    print(f"Removing {agents_id}")
-
+def remove_agents(agents_id, wazuh_script=True, wazuh_db_query=False):
     if agents_id:
         for agent_id in agents_id:
-           print(f"Removing {str(agent_id)}")
-           subprocess.call([f"{WAZUH_PATH}/bin/manage_agents", "-r", f"{agent_id}"], stdout=open(os.devnull, "w"), stderr=subprocess.STDOUT)
-           #result =  query_wdb(f"global delete-agent {str(agent_id)}")
+            if wazuh_script:
+                subprocess.call([f"{WAZUH_PATH}/bin/manage_agents", "-r", f"{agent_id}"], stdout=open(os.devnull, "w"),
+                                stderr=subprocess.STDOUT)
+            if wazuh_db_query:
+                result = query_wdb(f"global delete-agent {str(agent_id)}")
 
 def remove_all_agents():
     agent_ids_list = list_agents(['id'])

--- a/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
+++ b/deps/wazuh_testing/wazuh_testing/tools/wazuh_manager.py
@@ -1,19 +1,18 @@
 import os
 import subprocess
-import subprocess as sb
 from wazuh_testing.tools import WAZUH_PATH
 from wazuh_testing.wazuh_db import query_wdb
 
 
 def list_agents_ids():
-    wazuhdb_result = query_wdb("global get-all-agents last_id -1")
+    wazuhdb_result = query_wdb('global get-all-agents last_id -1')
     list_agents = [agent['id'] for agent in wazuhdb_result if not (0 == agent.get('id'))]
 
     return list_agents
 
 
 def remove_agents(agents_id, remove_type='wazuhdb'):
-    if remove_type not in ['wazuhdb','manage_agents']:
+    if remove_type not in ['wazuhdb', 'manage_agents']:
         raise ValueError("Invalid type of agent removal: %s" % remove_type)
 
     if agents_id:
@@ -26,5 +25,4 @@ def remove_agents(agents_id, remove_type='wazuhdb'):
 
 
 def remove_all_agents(remove_type):
-    agent_ids_list = list_agents_ids()
-    remove_agents(agent_ids_list, remove_type)
+    remove_agents(list_agents_ids(), remove_type)

--- a/tests/integration/test_wazuh_db/test_wazuh_db.py
+++ b/tests/integration/test_wazuh_db/test_wazuh_db.py
@@ -64,9 +64,8 @@ def clean_registered_agents():
 
 
 @pytest.fixture(scope="module")
-def retart_wazuh():
+def restart_wazuh():
     control_service('restart')
-
 
 
 @pytest.fixture(scope="function")
@@ -93,7 +92,7 @@ def pre_insert_agents():
                               for module_data, module_name in module_tests
                               for case in module_data]
                          )
-def test_wazuh_db_messages(clean_registered_agents, retart_wazuh, configure_sockets_environment, connect_to_sockets_module, test_case):
+def test_wazuh_db_messages(clean_registered_agents, restart_wazuh, configure_sockets_environment, connect_to_sockets_module, test_case):
     """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
 
     Parameters
@@ -117,7 +116,7 @@ def test_wazuh_db_messages(clean_registered_agents, retart_wazuh, configure_sock
             .format(index + 1, stage['stage'], expected, response)
 
 
-def test_wazuh_db_create_agent(clean_registered_agents, retart_wazuh, configure_sockets_environment, connect_to_sockets_module):
+def test_wazuh_db_create_agent(clean_registered_agents, restart_wazuh, configure_sockets_environment, connect_to_sockets_module):
     """Check that Wazuh DB creates the agent database when a query with a new agent ID is sent"""
     test = {"name": "Create agent",
             "description": "Wazuh DB creates automatically the agent's database the first time a query with a new agent"
@@ -125,11 +124,11 @@ def test_wazuh_db_create_agent(clean_registered_agents, retart_wazuh, configure_
             "test_case": [{"input": "agent 999 syscheck integrity_check_left",
                            "output": "err Invalid FIM query syntax, near 'integrity_check_left'",
                            "stage": "Syscheck - Agent does not exits yet"}]}
-    test_wazuh_db_messages(clean_registered_agents, configure_sockets_environment, connect_to_sockets_module, test['test_case'])
+    test_wazuh_db_messages(clean_registered_agents, restart_wazuh, configure_sockets_environment, connect_to_sockets_module, test['test_case'])
     assert os.path.exists(os.path.join(WAZUH_PATH, 'queue', 'db', "999.db"))
 
 
-def test_wazuh_db_chunks(clean_registered_agents, retart_wazuh, configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
+def test_wazuh_db_chunks(clean_registered_agents, restart_wazuh, configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
     """Check that commands by chunks work properly when agents amount exceed the response maximum size"""
 
     def send_chunk_command(command):

--- a/tests/integration/test_wazuh_db/test_wazuh_db.py
+++ b/tests/integration/test_wazuh_db/test_wazuh_db.py
@@ -9,6 +9,8 @@ import time
 import pytest
 import yaml
 from wazuh_testing.tools import WAZUH_PATH
+from wazuh_testing.tools.services import control_service
+from wazuh_testing.tools.wazuh_manager import remove_all_agents
 
 # Marks
 
@@ -56,7 +58,11 @@ def regex_match(regex, string):
     return re.match(regex, string)
 
 
-# Tests
+@pytest.fixture(scope="module")
+def clean_registered_agents():
+    control_service('restart')
+    remove_all_agents()
+
 
 @pytest.fixture(scope="function")
 def pre_insert_agents():
@@ -82,7 +88,7 @@ def pre_insert_agents():
                               for module_data, module_name in module_tests
                               for case in module_data]
                          )
-def test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_module, test_case: list):
+def test_wazuh_db_messages(clean_registered_agents, configure_sockets_environment, connect_to_sockets_module, test_case):
     """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
 
     Parameters
@@ -106,7 +112,7 @@ def test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_mod
             .format(index + 1, stage['stage'], expected, response)
 
 
-def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets_module):
+def test_wazuh_db_create_agent(clean_registered_agents, configure_sockets_environment, connect_to_sockets_module):
     """Check that Wazuh DB creates the agent database when a query with a new agent ID is sent"""
     test = {"name": "Create agent",
             "description": "Wazuh DB creates automatically the agent's database the first time a query with a new agent"
@@ -114,11 +120,11 @@ def test_wazuh_db_create_agent(configure_sockets_environment, connect_to_sockets
             "test_case": [{"input": "agent 999 syscheck integrity_check_left",
                            "output": "err Invalid FIM query syntax, near 'integrity_check_left'",
                            "stage": "Syscheck - Agent does not exits yet"}]}
-    test_wazuh_db_messages(configure_sockets_environment, connect_to_sockets_module, test['test_case'])
+    test_wazuh_db_messages(clean_registered_agents, configure_sockets_environment, connect_to_sockets_module, test['test_case'])
     assert os.path.exists(os.path.join(WAZUH_PATH, 'queue', 'db', "999.db"))
 
 
-def test_wazuh_db_chunks(configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
+def test_wazuh_db_chunks(clean_agent_registered, configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
     """Check that commands by chunks work properly when agents amount exceed the response maximum size"""
 
     def send_chunk_command(command):

--- a/tests/integration/test_wazuh_db/test_wazuh_db.py
+++ b/tests/integration/test_wazuh_db/test_wazuh_db.py
@@ -60,7 +60,8 @@ def regex_match(regex, string):
 
 @pytest.fixture(scope="module")
 def clean_registered_agents():
-    remove_all_agents()
+    remove_all_agents('wazuhdb')
+    time.sleep(5)
 
 
 @pytest.fixture(scope="module")
@@ -92,7 +93,7 @@ def pre_insert_agents():
                               for module_data, module_name in module_tests
                               for case in module_data]
                          )
-def test_wazuh_db_messages(clean_registered_agents, restart_wazuh, configure_sockets_environment, connect_to_sockets_module, test_case):
+def test_wazuh_db_messages(restart_wazuh, clean_registered_agents, configure_sockets_environment, connect_to_sockets_module, test_case):
     """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
 
     Parameters
@@ -116,7 +117,7 @@ def test_wazuh_db_messages(clean_registered_agents, restart_wazuh, configure_soc
             .format(index + 1, stage['stage'], expected, response)
 
 
-def test_wazuh_db_create_agent(clean_registered_agents, restart_wazuh, configure_sockets_environment, connect_to_sockets_module):
+def test_wazuh_db_create_agent(restart_wazuh, clean_registered_agents, configure_sockets_environment, connect_to_sockets_module):
     """Check that Wazuh DB creates the agent database when a query with a new agent ID is sent"""
     test = {"name": "Create agent",
             "description": "Wazuh DB creates automatically the agent's database the first time a query with a new agent"
@@ -128,7 +129,7 @@ def test_wazuh_db_create_agent(clean_registered_agents, restart_wazuh, configure
     assert os.path.exists(os.path.join(WAZUH_PATH, 'queue', 'db', "999.db"))
 
 
-def test_wazuh_db_chunks(clean_registered_agents, restart_wazuh, configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
+def test_wazuh_db_chunks(restart_wazuh, clean_registered_agents, configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
     """Check that commands by chunks work properly when agents amount exceed the response maximum size"""
 
     def send_chunk_command(command):

--- a/tests/integration/test_wazuh_db/test_wazuh_db.py
+++ b/tests/integration/test_wazuh_db/test_wazuh_db.py
@@ -60,8 +60,13 @@ def regex_match(regex, string):
 
 @pytest.fixture(scope="module")
 def clean_registered_agents():
-    control_service('restart')
     remove_all_agents()
+
+
+@pytest.fixture(scope="module")
+def retart_wazuh():
+    control_service('restart')
+
 
 
 @pytest.fixture(scope="function")
@@ -88,7 +93,7 @@ def pre_insert_agents():
                               for module_data, module_name in module_tests
                               for case in module_data]
                          )
-def test_wazuh_db_messages(clean_registered_agents, configure_sockets_environment, connect_to_sockets_module, test_case):
+def test_wazuh_db_messages(clean_registered_agents, retart_wazuh, configure_sockets_environment, connect_to_sockets_module, test_case):
     """Check that every input message in wazuh-db socket generates the adequate output to wazuh-db socket
 
     Parameters
@@ -112,7 +117,7 @@ def test_wazuh_db_messages(clean_registered_agents, configure_sockets_environmen
             .format(index + 1, stage['stage'], expected, response)
 
 
-def test_wazuh_db_create_agent(clean_registered_agents, configure_sockets_environment, connect_to_sockets_module):
+def test_wazuh_db_create_agent(clean_registered_agents, retart_wazuh, configure_sockets_environment, connect_to_sockets_module):
     """Check that Wazuh DB creates the agent database when a query with a new agent ID is sent"""
     test = {"name": "Create agent",
             "description": "Wazuh DB creates automatically the agent's database the first time a query with a new agent"
@@ -124,7 +129,7 @@ def test_wazuh_db_create_agent(clean_registered_agents, configure_sockets_enviro
     assert os.path.exists(os.path.join(WAZUH_PATH, 'queue', 'db', "999.db"))
 
 
-def test_wazuh_db_chunks(clean_agent_registered, configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
+def test_wazuh_db_chunks(clean_registered_agents, retart_wazuh, configure_sockets_environment, connect_to_sockets_module, pre_insert_agents):
     """Check that commands by chunks work properly when agents amount exceed the response maximum size"""
 
     def send_chunk_command(command):

